src/protocol/mod.rs

```diff
@@ -78,6 +96,12 @@ mod str_bytes {
         }
     }
 
+    impl Display for StrBytes {
+        fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+            std::fmt::Display::fmt(&**self, f)
```

Comment:
I was a little confused what this was doing.
Eventually I figured out its using the Deref impl to call `.as_str()`.

So to make it a little more obvious maybe call `.as_str()` directly?
`std::fmt::Display::fmt(self.as_str(), f)`

Also, we dont have any unit tests for StrBytes, and I think just adding one would go a long way in ensuring basic functionality works:
```rust
#[test]
fn strbytes_display() {
    let string = StrBytes::from_static_str("foobar");
    assert_eq!(format!("{string}"), "foobar".to_owned());
}
```

But these are both minor things, I'm happy to approve now.